tracker/tracker/src/main/app/nodes.ts

```diff
@@ -57,6 +60,23 @@ export default class Nodes {
     }
     return id;
   }
+  cleanTree() {
+    // sadly we keep empty items in array here resulting in some memory still being used
+    // but its still better than keeping dead nodes or undef elements
+    // plus we keep our index positions for new/alive nodes
+    // performance test: 3ms for 30k nodes with 17k dead ones
+    for (let i = 0; i < this.nodes.length; i++) {
+      const node = this.nodes[i];
+      if (node === undefined) {
+        delete this.nodes[i];
```

Comment:
why delete if it is already undefined?

Reply:
ok, I see. just replace `this.nodes[id] = undefined;` to `delete this.nodes[id]` in the `unregisterNode` fn